libs/sysplugins/smarty_internal_compile_private_special_variable.php

```diff
@@ -76,7 +77,8 @@ public function compile($args, Smarty_Internal_TemplateCompilerBase $compiler, $
                         $compiler->trigger_template_error("(secure mode) super globals not permitted");
                         break;
                     }
-                    $compiled_ref = '$_' . strtoupper($variable);
+                    $variableToUpper = strtr($variable, 'abcdefghijklmnopqrstuvwxyz', 'ABCDEFGHIJKLMNOPQRSTUVWXYZ');
```

Comment:
```suggestion
                    $variableUpper = strtr($variable, 'abcdefghijklmnopqrstuvwxyz', 'ABCDEFGHIJKLMNOPQRSTUVWXYZ');
```

Reply:
(Thanks for spotting this mistake, @Alexkurd!)

Reply:
You're welcome. I've repeated the error earlier with AZ locale.